pkgs/development/libraries/drumstick/default.nix

```diff
@@ -18,6 +18,20 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     substituteInPlace library/rt/backendmanager.cpp --subst-var out
+
+    # https://sourceforge.net/p/drumstick/bugs/39/
+    substituteInPlace drumstick-alsa.pc.in \
+      --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
+      --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+    substituteInPlace drumstick-file.pc.in \
+      --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
+      --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+    substituteInPlace drumstick-rt.pc.in \
+      --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
+      --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+    substituteInPlace drumstick-widgets.pc.in \
+      --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
+      --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
```

Comment:
```suggestion
    for file in drumstick-alsa.pc.in drumstick-file.pc.in drumstick-rt.pc.in drumstick-widgets.pc.in; do
      substituteInPlace $file \
        --replace '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@ \
        --replace '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
    do
```

Reply:
(For this package specifically the fix is already committed and upstream promised a release soon so I’ll try and wait for that.)

Reply:
v2.6.1 released:
https://sourceforge.net/p/drumstick/news/2022/05/drumstick-libraries-261-released-/

This problem with the pkg-config files should be solved, and also the need of a patch for locating the drumstick backends when CMAKE_INSTALL_LIBDIR has an absolute path.

Reply:
> (For this package specifically the fix is already committed and upstream promised a release soon so I’ll try and wait for that.)

More than one month after a new upstream release has been published, nobody has bothered to upgrade the [NixOS package](https://repology.org/project/drumstick/versions), neither @alexshpilkin has updated his patch. 
cc: @dotlambda: this is why I think that flathub/flatpak and friends are fixing a fundamental flaw of conventional Linux packaging.

Reply:
@pedrolcl You can see I bet on fixing all of the brokenness at once. The delay is because I lost: among the problematic packages is `google-cloud-cpp`, where the maintainer, commendably, went for a substantial refactor rather than patch over the couple dozen autogenerated files as they had them, but now I need to bump it three versions and figure out C++ ABI compatibility and each failing compile takes several hours on my scruffy i5. I haven’t yet got back to that unpleasantness, that’s all. I know it’s a poor response to the quick fixes I received from yourself and quite a number of others, but my emotional capacity has been extremely limited as of late. I am sorry, for what it’s worth.

(The benefits and drawbacks of having a packager intermediate between the author and the user have been extensively flamed [elsewhere][1]—hell if I know how I’m going to patch glslang and shaderc, which decided to *vendor* broken headers I otherwise already fixed—but in any case Nixpkgs, unlike Debian or Fedora or even FreeBSD ports, does not have a coherent team of package maintainers, so it’s perfectly normal for *anyone* to submit a version bump. If there hasn’t been one, then nobody either had a need for it or knew enough to make it. That is the price for having what’s probably the most extensive collection of packages in existence, and it’s not going to be acceptable to everyone.)

[1]: http://web.archive.org/web/20210220023308if_/https://blogs.gentoo.org/mgorny/2021/02/19/the-modern-packagers-security-nightmare/

Reply:
@pedrolcl While on Flathub close to all leaf packages might be up to date, the same can definitely not be said about their dependencies. That's where the security risk lies. Btw Nixpkgs is about 85% up to date according to Repology.

Reply:
> > (For this package specifically the fix is already committed and upstream promised a release soon so I’ll try and wait for that.)
> 
> More than one month after a new upstream release has been published, nobody has bothered to upgrade the [NixOS package](https://repology.org/project/drumstick/versions), neither @alexshpilkin has updated his patch. cc: @dotlambda: this is why I think that flathub/flatpak and friends are fixing a fundamental flaw of conventional Linux packaging.

if you use some package which does not have a maintainer then i recommend adding yourself as a maintainer for it and submitting a pr updating it, that is the best thing you can do.

Reply:
> if you use some package which does not have a maintainer then i recommend adding yourself as a maintainer for it and submitting a pr updating it, that is the best thing you can do.

If nobody cares about drumstick in NixOS, then there is an easier solution: remove it. The problem is that other projects depend on it. I hope that  @orivej cares about VMPK and kmetronome.

I don't use NixOS myself, but I am a developer of the package we are talking about. I have no plans of packaging my projects for all and every Linux distro out there. For me Flathub is a perfect solution: I only have to care about a single repository, and I can offer all my projects even if they are all niche products.

@dotlambda: another problem with the Linux distros is the niche applications. They are often excluded from the repositories. There are 232987 projects [known to repology](https://repology.org/repositories/statistics/total) (because they are at least in one repository). The the topmost repository by number of projects is nixpkgs unstable with 72002 projects (31% of the total). 

Here is an example of a niche application: [dmidiplayer](https://repology.org/project/dmidiplayer/versions) available for FreeBSD, MacOS, and Windows, but excluded of most Linux distros.

Reply:
> #179594

When I saw that NixOS was patching my library sources, I've [included another fix](https://github.com/pedrolcl/drumstick/commit/76da86e20f83a4216a065c3213397a820f8efba1) in the release 2.6.1

I don't know if the new version with the old patch that is still being applied works, but any way you are risking that the patch don't apply in a future release.

Reply:
> I don't use NixOS myself, but I am a developer of the package we are talking about. I have no plans of packaging my projects for all and every Linux distro out there.

You don't have to. That's the job of packagers. If users want a package for a specific distro, let them request it to the distro maintainers

Reply:
> I don't know if the new version with the old patch that is still being applied works, but any way you are risking that the patch don't apply in a future release.

That can always happen and then the patch needs to be adjusted in the next update. Not a problem and pretty normal.